diff --git a/node_modules/react-native-reanimated/Common/cpp/reanimated/LayoutAnimations/LayoutAnimationsManager.cpp b/node_modules/react-native-reanimated/Common/cpp/reanimated/LayoutAnimations/LayoutAnimationsManager.cpp
index d7d3e8b..ffa555c 100644
--- a/node_modules/react-native-reanimated/Common/cpp/reanimated/LayoutAnimations/LayoutAnimationsManager.cpp
+++ b/node_modules/react-native-reanimated/Common/cpp/reanimated/LayoutAnimations/LayoutAnimationsManager.cpp
@@ -37,7 +37,10 @@ void LayoutAnimationsManager::configureAnimationBatch(
     if (config == nullptr) {
       continue;
     }
-    sharedTransitionGroups_[sharedTransitionTag].push_back(tag);
+    auto &group = sharedTransitionGroups_[sharedTransitionTag];
+    if (std::find(group.begin(), group.end(), tag) == group.end()) {
+      group.push_back(tag);
+    }
     viewTagToSharedTag_[tag] = sharedTransitionTag;
     getConfigsForType(SHARED_ELEMENT_TRANSITION)[tag] = config;
     if (type == SHARED_ELEMENT_TRANSITION) {
